ethz-asl/moma
# Projects/piloting feature/smb integration

This PR integrates the SMB and Panda simulations together. 
In order to run the simulation the SMB stack needs to be cloned. This is reflected in the `moma_piloting.repos` which should be merged using `vcstool`. 

### How to test
1. merge dependencies from source root with `vcs import --input moma/moma_piloting.repos`
2. update dependencies of the smb stack through `rosdep`: `rosdep install --from-paths src --ignore-src --skip-keys="pinocchio" -r -y`
  When I did this some packages are not found. Then I had to iterate the build process and install all missing packages with `sudo apt-get install ros-noetic-<pkgname>` until all packages were resolved. This is an annoying procedure, that we could improve changing the smb stack (we depend on a custom fork of the public repo, we can do that)
3. build the `piloting_demo` package (see above point if something goes wrong)
4. launch simulation (arm + base + localization + planner + base control): `roslaunch piloting_demo superpanda_sim.launch`

The last step should spawn something like this in rviz (note the additional navigation pluing):
 
![navigation_sim](https://user-images.githubusercontent.com/26412496/143591049-21d36783-2854-472d-9eb8-bc694bda0db0.png)

Make sure the correct odom frame is set in the navigation panel. After that, one can click on `Edit` to move the 2d interactive marker to set a new base pose, and after that `Start Planning` to generate a global path which is then followed by the base (tracking is not perfect and the base will stop moving in a certain tolerance).

More info about the SMB navigation and control stack here: https://ethz-robotx.github.io/SuperMegaBot/
''
